docker/Dockerfile

```diff
@@ -1,40 +1,70 @@
+# set path
+ARG PRODUCTION_PATH="/usr/local/app"
+
+# Use official node image as the base image
+FROM node:16-bullseye-slim AS builder
+
+# needed to use PRODUCTION_PATH before the first FROM
+ARG PRODUCTION_PATH
+
+# add the source code to app
+COPY ./ $PRODUCTION_PATH
+
+# set workdir
+WORKDIR $PRODUCTION_PATH
+
+# install for production and cleanup
+RUN npm run deployment
+
+####  building done  ####
+#########################
+####  prod container ####
 # Use official node image as the base image, important to be on debian based for apt-get
 FROM node:16-bullseye-slim
 
+# needed to use PRODUCTION_PATH before the first FROM
+ARG PRODUCTION_PATH
+
 # Disable interactive frontend to prevent install dialogues
 ARG DEBIAN_FRONTEND=noninteractive
+# set production patch
 
-# install nginx
-RUN apt-get update && apt-get upgrade -y && apt-get install nginx -y
+WORKDIR $PRODUCTION_PATH
 
-# add the source code to app
-COPY ./ /usr/local/app/
-
-# set workdir
-WORKDIR /usr/local/app
+# install necessary packages
+RUN apt-get update && \
+    apt-get upgrade -y && \
+    apt-get install --no-install-recommends \
+    nginx \
+    ca-certificates -y && \
+    rm -rf /var/lib/apt/lists/*
 
-# setup and install all the dependencies
-RUN npm run setup
+# copy frontend
+COPY --chown=www-data:www-data --from=builder $PRODUCTION_PATH/frontend/dist $PRODUCTION_PATH/frontend/dist
 
-# Generate the build of the application
-RUN npm run build:deployment
+# copy shared
+COPY --chown=www-data:www-data --from=builder $PRODUCTION_PATH/shared/dist $PRODUCTION_PATH/shared/dist
+COPY --chown=www-data:www-data --from=builder $PRODUCTION_PATH/shared/node_modules $PRODUCTION_PATH/shared/node_modules
+COPY --chown=www-data:www-data --from=builder $PRODUCTION_PATH/shared/package.json $PRODUCTION_PATH/shared
```

Comment:
Why do you need this file?

Reply:
when I comment out shared/package.json comes smth. like:
```
Error [ERR_MODULE_NOT_FOUND]: Cannot find package '/usr/local/app/backend/node_modules/digital-fuesim-manv-shared/' imported from /usr/local/app/backend/dist/src/exercise/websocket.js
```
when I comment out backend/package.json comes smth. like:
```
 import 'dotenv/config';
digital-fuesim-manv    | ^^^^^^
digital-fuesim-manv    |
digital-fuesim-manv    | SyntaxError: Cannot use import statement outside a module
```

Reply:
Ah, right. Completely forgot about that.